src/ReduxCanvasV4.js

```diff
@@ -0,0 +1,61 @@
+import Immutable from 'immutable'
+import React from 'react'
+import { Provider, connect } from 'react-redux'
+import createStore from './createStore'
+import Pixel from './Pixel'
+
+// Reducer
+const store = createStore((state = Immutable.Map(), action) => {
+  if (action.type === 'TOGGLE') {
+    const key = action.i + ',' + action.j
+    return state.set(key, !state.get(key))
+  }
+  return state
+})
+
+const ACTIVE_PROPS = { active: true }
+const INACTIVE_PROPS = { active: false }
+
+// Connected pixel
+const ConnectedPixel = connect(
+  (initialState, initialProps) => {
+    const { i, j } = initialProps
+    return state => {
+      const active = state.get(i + ',' + j) || false
+      return active ? ACTIVE_PROPS : INACTIVE_PROPS
+    }
+  },
+  (initialState, initialProps) => (dispatch) => {
```

Comment:
Oh that’s right. Thanks for catching this! 😄

We’re just using the factory function just so that `doDispatchPropsDependOnOwnProps` is `false` and can let `react-redux` do its optimization.